### PR TITLE
Add fetching the well known in embedded mode.

### PR DESCRIPTION
**Pull request by toger5**
This is used to load the focus from the well known in elment-call.

## Checklist

-   [x] Tests written for new code (and old code if feasible).
-   [x] New or updated `public`/`exported` symbols have accurate [TSDoc](https://tsdoc.org/) documentation.
-   [x] Linter and other CI checks pass.
-   [x] Sign-off given on the changes (see [CONTRIBUTING.md](https://github.com/matrix-org/matrix-js-sdk/blob/develop/CONTRIBUTING.md)).

### CODE REVIEW

File: src/matrixrtc/MatrixRTCSession.ts [hunk 1]
```diff
@@ -825,7 +825,8 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
 
         const myCallMemberEvent = roomState.getStateEvents(EventType.GroupCallMemberPrefix, localUserId) ?? undefined;
         const content = myCallMemberEvent?.getContent() ?? {};
-        const legacy = "memberships" in content || this.useLegacyMemberEvents;
+        const legacy =
+            (("memberships" in content && content.membership?.length) ?? 0 > 0) || this.useLegacyMemberEvents;
```
Comment (AndrewFerr) [inline src/matrixrtc/MatrixRTCSession.ts:829]: Is the second `membership` meant to be `memberships` of a `*.call.member` event?
```suggestion
            (("memberships" in content && content.memberships?.length) ?? 0 > 0) || this.useLegacyMemberEvents;
```

Comment (toger5) [reply src/matrixrtc/MatrixRTCSession.ts:829]: of course!! thx

**Comment by dbkr** [comment]
Unassigning myself for now as it looks like this is still in-progress and getting (force) pushes.

File: src/matrixrtc/MatrixRTCSession.ts
Comment (AndrewFerr) [inline src/matrixrtc/MatrixRTCSession.ts:1]: IMO the changes on this file should be in their own PRs, as they aren't related to well-known fetching.

File: src/matrixrtc/MatrixRTCSession.ts [hunk 2]
```diff
@@ -825,7 +825,8 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
 
         const myCallMemberEvent = roomState.getStateEvents(EventType.GroupCallMemberPrefix, localUserId) ?? undefined;
         const content = myCallMemberEvent?.getContent() ?? {};
-        const legacy = "memberships" in content || this.useLegacyMemberEvents;
+        const legacy =
+            (("memberships" in content && content.memberships?.length) ?? 0 > 0) || this.useLegacyMemberEvents;
```
Comment (AndrewFerr) [inline src/matrixrtc/MatrixRTCSession.ts:829]: This will fallback to the legacy format only if the _current user's_ `*.call.member` state event uses the legacy format.  Should it fallback if _any_ user's `*.call.member` state uses the legacy format?

Comment (toger5) [reply src/matrixrtc/MatrixRTCSession.ts:829]: I was thinking about this as well. But then decided that we can make the transition a bit more abrupt and only added this usecase to save time. I saw your PR however and like it more then this. So I will remove this change from this pr.

File: src/matrixrtc/MatrixRTCSession.ts
Comment (toger5) [reply src/matrixrtc/MatrixRTCSession.ts:1]: moved them out of here. I saw you have a PR covering those. Commented on this pr.

**Comment by toger5** [comment]
> Unassigning myself for now as it looks like this is still in-progress and getting (force) pushes.

sorry for the force push.

Now it is ready to review.

We removed unrelated changes.

File: src/client.ts [hunk 3]
```diff
@@ -7410,7 +7410,7 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         return this.http.authedRequest(Method.Post, path, undefined, undefined, { prefix: "" });
     }
 
-    private async fetchClientWellKnown(): Promise<void> {
+    public async fetchClientWellKnown(): Promise<void> {
```
Comment (AndrewFerr) [inline src/client.ts:7413]: How about `protected`?  That would avoid altering the public API of `MatrixClient`.
```suggestion
    protected async fetchClientWellKnown(): Promise<void> {
```

**Review by AndrewFerr — APPROVED** [review]

**Review by dbkr — APPROVED** [review]
